cdap-ui/app/cdap/components/Administration/AdminConfigTabContent/NamespacesAccordion/index.js

```diff
@@ -91,15 +113,39 @@ export default class NamespacesAccordion extends Component {
               datasetCount
             } = getCustomAppPipelineDatasetCounts(entities);
 
+            let namespaceIsHighlighted = false;
+
+            if (currentNamespaces.length && currentNamespaces.indexOf(namespace.name) === -1) {
+              namespaceIsHighlighted = true;
+              hasNewNamespaces = true;
+            }
+
             namespacesInfo.push({
               name: namespace.name,
               pipelineCount,
               customAppCount,
-              datasetCount
+              datasetCount,
+              highlighted: namespaceIsHighlighted
             });
+
             this.setState({
               namespacesInfo,
-              loading: false
+              loading: false,
+              viewAll: hasNewNamespaces || this.state.viewAll
```

Comment:
so if there's a new namespace, it will always expand the viewAll?

Also, what are the different function for `namespaceIsHighlighted` vs `hasNewNamespaces`? seems like if it has new namespace, then it would be highlighted

Reply:
Yes, it will always expand (if there are > 10 existing namespaces). The reason for this is that if we want to check the index of the newly added namespace in the list of namespaces, we have to wait for all the rows to render, since we have no way to know which row will render first. This might cause performance issues, which might not be worth it if the only reason we're waiting is to decide whether to set the `viewAll` or not.

The `namespaceIsHighlighted` is to check to see whether the current namespace should be highlighted (i.e. is new). The `hasNewNamespaces` is to check whether any new namespace were added, to set the `viewAll`. We can only add one namespace at a time, so I guess I can simplify this, but in `SystemPrefsAccordion` where I have similar logic, the user can add multiple new prefs at a time, so the distinction between the two variables is more clear there.